### PR TITLE
Add Meter component for bounded measurement semantics

### LINKED ISSUE #482: New component: Meter (bounded measurement gauge, distinct from Progress)
## Source
[bits-ui Meter](https://bits-ui.com/docs/components/meter). cinder has `progress` but no `meter`.

## Gap
`progress` semantically represents **task completion** — a value that only moves toward done (file upload, form completion). A **meter** represents a **static measurement within a known range** that fluctuates: CPU usage, battery level, disk space, memory pressure, rate-limit consumption. These are different ARIA roles (`role="progressbar"` vs `role="meter"`) and different semantics; reusing `progress` for gauges is incorrect for assistive tech.

## Proposal
Add a `meter` component.

- `role="meter"`
- Props: `value` (default 0), `min` (default 0), `max` (default 100)
- `aria-valuetext` support for compound readouts (e.g. "50% (6 hours) remaining")
- `aria-label` / `aria-labelledby` wiring
- Data attributes `data-value` / `data-min` / `data-max` for styling
- Optional threshold/segment styling (low/optimal/high) is a reasonable cinder addition given our token system

## Notes
Keep the visual language aligned with `progress` but make the semantic + ARIA distinction the whole point of the component. Worth a short doc note steering consumers between `progress` and `meter`.

### PR DESCRIPTION
**Pull request by stevekinney**
## Why
`Progress` models task completion, but issue #482 needs a semantically correct bounded measurement gauge for values like battery, CPU, and quota usage. This adds a dedicated `Meter` so assistive tech gets `role="meter"` semantics instead of `progressbar` semantics.

## What changed
- Added a new public `Meter` component (`@lostgradient/cinder/meter`) with:
  - `value`, `min`, `max` bounds handling and clamped rendering
  - `low`, `high`, `optimum` thresholds with segmented low/optimum/high track regions
  - `ariaLabel` / `ariaLabelledby` naming support and `ariaValueText` override
  - stable state/data attributes (`data-cinder-state`, `data-value`, `data-min`, `data-max`)
  - development warnings for invalid ranges, non-finite thresholds, and out-of-range values
- Added meter styles, tests, generated schema/variables/examples artifacts, README, and a11y doc.
- Added playground examples for basic usage, custom ranges, and segmented thresholds.
- Wired package surfaces and metadata:
  - root exports in `src/index.ts`
  - style aggregation in `src/styles/components.css`
  - package exports and manifest generation outputs (`package.json`, `components.json`)
- Updated `Progress` metadata guidance to explicitly steer bounded measurement use cases to `Meter`.

## Notes for reviewers
The segmented regions are intentionally rendered in-track (not just state tokens) while preserving non-color semantics through explicit ARIA value text and machine-readable data attributes.

Fixes: #482

---

> [!NOTE]
> **Low Risk**
> Additive alpha component with tests and docs; only metadata on Progress changes, no runtime behavior change to existing consumers.
> 
> **Overview**
> Adds a new **`Meter`** feedback component (`@lostgradient/cinder/meter`) for bounded readings (battery, quota, CPU, etc.) using **`role="meter"`** instead of task-completion **`progressbar`** semantics.
> 
> The implementation clamps **`value`** to **`min`/`max`**, optional **`low`/`high`/`optimum`** thresholds with in-track low/optimum/high segments and **`data-cinder-state`**, plus **`ariaLabel`/`ariaLabelledby`** and optional **`ariaValueText`**. Dev warnings cover invalid ranges and missing accessible names.
> 
> Packaging and docs are wired through root exports, **`components.json`**, package subpath exports, global styles, schema/variables/examples, README and a11y notes, playground examples, and a broad unit test suite. **`Progress`** metadata now points bounded-gauge use cases at **`Meter`**. Playground discovery tests expect **`meter`** in the sidebar (ceiling 139). Minor version bump via changeset.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 91259189a65cb0e404a86431bcaa11b210bfa515. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/components/src/components/meter/meter.svelte [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        style:width="{lowPercent}%"
+      ></div>
+      <div
+        class="cinder-meter__segment cinder-meter__segment--optimum"
+        style:width="{optimumPercent}%"
+      ></div>
+      <div
+        class="cinder-meter__segment cinder-meter__segment--high"
+        style:width="{highPercent}%"
+      ></div>
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Implemented. Segment tone mapping now follows optimum direction, so low/high/optimum bands stay semantically aligned when optimum is outside the middle range.

File: packages/components/src/components/meter/meter.css [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  .cinder-meter__fill {
+    position: absolute;
+    inset-block: 0;
+    inset-inline-start: 0;
+    inline-size: 100%;
+    background: var(--cinder-accent);
+    border-radius: inherit;
+    transform-origin: center left;
+    transform: scaleX(var(--_cinder-meter-progress, 0));
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.css:64]: Good catch. I added an RTL override so the fill transform origin switches to the inline end in RTL, keeping fill growth aligned with the minimum-to-maximum direction.

File: packages/components/src/components/meter/meter.svelte [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      );
+    }
+  });
+</script>
+
+<div
+  class={classNames('cinder-meter', className)}
+  role="meter"
+  aria-label={ariaLabel}
+  aria-labelledby={ariaLabelledby}
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Updated. Empty-string `ariaLabel` and `ariaLabelledby` values are now normalized to `undefined`, so empty ARIA name attributes are omitted.

File: packages/components/src/components/meter/meter.svelte [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      ></div>
+      <div
+        class="cinder-meter__segment cinder-meter__segment--optimum"
+        style:width="{optimumPercent}%"
+      ></div>
+      <div
+        class="cinder-meter__segment cinder-meter__segment--high"
+        style:width="{highPercent}%"
+      ></div>
+    </div>
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Adjusted. Threshold bands and `data-cinder-state` are now opt-in and only render when at least one threshold prop (`low`, `high`, or `optimum`) is provided.

File: packages/components/src/components/meter/meter.svelte [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  const effectiveOptimum = $derived(
+    optimum === undefined || !Number.isFinite(optimum)
+      ? effectiveMin + range / 2
+      : Math.max(effectiveMin, Math.min(effectiveMax, optimum)),
+  );
+  const normalized = $derived((clampedValue - effectiveMin) / range);
+  const progressScale = $derived(normalized);
+  const percent = $derived(Math.round(progressScale * 100));
+  const defaultValueText = $derived(`${percent}%`);
+  const valueText = $derived(ariaValueText ?? defaultValueText);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Fixed. The component no longer generates a normalized percent `aria-valuetext` by default; `aria-valuetext` is now only set when `ariaValueText` is provided.

File: packages/components/src/components/meter/meter.svelte [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+  const hasFiniteValue = $derived(Number.isFinite(value));
+  const rawValue = $derived(hasFiniteValue ? value : effectiveMin);
+  const clampedValue = $derived(Math.max(effectiveMin, Math.min(effectiveMax, rawValue)));
+
+  const lowBoundary = $derived(
+    low === undefined || !Number.isFinite(low)
+      ? effectiveMin + range / 3
+      : Math.max(effectiveMin, Math.min(effectiveMax, low)),
+  );
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte:60]: Implemented. When thresholds are omitted, the meter now remains neutral: no inferred threshold bands and no derived low/high state.

File: packages/components/src/components/meter/meter.css [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    opacity: 0.32;
+  }
+
+  .cinder-meter__segment--optimum {
+    background: var(--cinder-success);
+    opacity: 0.28;
+  }
+
+  .cinder-meter__segment--high {
+    background: var(--cinder-danger);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.css]: Addressed. Segment coloring is now derived from optimum direction, so high-is-better and low-is-better configurations keep the track semantics consistent with state.

File: packages/components/src/components/meter/meter.svelte [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+  const lowBoundary = $derived(
+    low === undefined || !Number.isFinite(low)
+      ? effectiveMin + range / 3
+      : Math.max(effectiveMin, Math.min(effectiveMax, low)),
+  );
+  const highBoundary = $derived(
+    high === undefined || !Number.isFinite(high)
+      ? effectiveMin + (range * 2) / 3
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Implemented. Omitted `high`/`low` thresholds now stay pinned to range edges (`max`/`min`) instead of synthesized thirds, so we no longer infer an undeclared danger boundary.

File: packages/components/src/components/meter/meter.svelte [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    typeof ariaLabelledby === 'string' && ariaLabelledby.trim().length > 0
+      ? ariaLabelledby
+      : undefined,
+  );
+
+  const meterState = $derived.by<MeterState | undefined>(() => {
+    if (!hasThresholds) return undefined;
+    if (effectiveOptimum < segmentLow) {
+      if (clampedValue <= segmentLow) return 'optimum';
+      return 'high';
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte:95]: Fixed. For low-is-better and high-is-better configurations, the middle band now remains distinct from the worst band, so the caller-provided `high` threshold continues to influence state.

File: packages/playground/src/examples/meter/basic.example.svelte [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  export const title = 'Basic meter';
+  export const description = 'Bounded measurement for a fluctuating value.';
+</script>
+
+<script lang="ts">
+  import { Meter } from '@lostgradient/cinder/meter';
+</script>
+
+<div style="display: flex; flex-direction: column; gap: 0.5rem;">
+  <Meter value={32} ariaLabel="Battery level" />
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/playground/src/examples/meter/basic.example.svelte:11]: Updated. I raised the discovery sidebar ceiling gate and added `meter` as a positive anchor in `discover.test.ts`.

File: packages/components/src/components/meter/meter.svelte [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+          style:width="{optimumPercent}%"
+        ></div>
+        <div
+          class={classNames(
+            'cinder-meter__segment',
+            'cinder-meter__segment--band-high',
+            `cinder-meter__segment--state-${highSegmentTone}`,
+          )}
+          style:width="{highPercent}%"
+        ></div>
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte:202]: Implemented. I switched all threshold segment bindings to `style:inline-size` so the runtime style matches the CSS sizing property and segments render correctly.

File: packages/components/src/components/meter/meter.test.ts [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+      min: 0,
+      max: 100,
+      low: 30,
+      high: 70,
+      ariaLabel: 'Quota usage',
+    });
+    const segments = container.querySelectorAll('.cinder-meter__segment');
+
+    expect(segments).toHaveLength(3);
+  });
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.test.ts:112]: Added. The segment test now asserts the rendered inline-size percentages for low/optimum/high bands so zero-size regressions are caught.

File: packages/components/src/components/meter/meter.svelte [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+    if (!hasThresholds) return undefined;
+    if (effectiveOptimum < segmentLow) {
+      if (clampedValue <= segmentLow) return 'optimum';
+      if (clampedValue <= segmentHigh) return 'low';
+      return 'high';
+    }
+    if (effectiveOptimum > segmentHigh) {
+      if (clampedValue < segmentLow) return 'low';
+      if (clampedValue <= segmentHigh) return 'high';
+      return 'optimum';
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Implemented. I corrected the high-is-better mapping so values below `low` are now the worst (`high`) state and the `low..high` middle band is the warning (`low`) state; segment tones were updated to match.

File: packages/components/package.json [hunk 14]
```diff
@@ -2604,6 +2604,34 @@
       "import": "./src/components/message/message.examples.json",
       "default": "./src/components/message/message.examples.json"
     },
+    "./meter": {
+      "types": "./dist/components/meter/index.d.ts",
+      "browser": "./src/components/meter/index.ts",
+      "node": "./dist/server/components/meter/index.js",
+      "svelte": "./src/components/meter/index.ts",
+      "default": "./dist/components/meter/index.js"
+    },
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/package.json:2613]: Added. I created a changeset entry for `@lostgradient/cinder` to capture the new public `meter` export in release/versioning.

File: packages/components/src/components/meter/meter.svelte [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+    if (effectiveOptimum > segmentHigh) {
+      if (clampedValue < segmentLow) return 'high';
+      if (clampedValue <= segmentHigh) return 'low';
+      return 'optimum';
+    }
+    if (clampedValue < segmentLow) return 'low';
+    if (clampedValue <= segmentHigh) return 'optimum';
+    return 'high';
+  });
+  const lowSegmentTone = $derived<MeterState>(optimumDirection === 'high' ? 'high' : 'low');
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Implemented. In low-is-better mode (`optimum <= low`), the first segment now uses the optimum tone so the threshold guide matches fill-state semantics.

File: packages/components/src/components/meter/meter.svelte [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+  const segmentLow = $derived(Math.min(lowBoundary, highBoundary));
+  const segmentHigh = $derived(Math.max(lowBoundary, highBoundary));
+  const effectiveOptimum = $derived(
+    optimum === undefined || !Number.isFinite(optimum)
+      ? effectiveMin + range / 2
+      : Math.max(effectiveMin, Math.min(effectiveMax, optimum)),
+  );
+  const optimumDirection = $derived.by<OptimumDirection>(() => {
+    if (effectiveOptimum < segmentLow) return 'low';
+    if (effectiveOptimum > segmentHigh) return 'high';
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Fixed. Boundary equality is now directional: `optimum <= low` is low-is-better and `optimum >= high` is high-is-better, so explicit-threshold boundary values follow native meter semantics.

File: packages/components/src/components/meter/meter.svelte [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+    }
+    if (clampedValue < segmentLow) return 'low';
+    if (clampedValue <= segmentHigh) return 'optimum';
+    return 'high';
+  });
+  const lowSegmentTone = $derived<MeterState>(optimumDirection === 'high' ? 'high' : 'low');
+  const optimumSegmentTone = $derived<MeterState>(
+    optimumDirection === 'mid' ? 'optimum' : 'low',
+  );
+  const highSegmentTone = $derived<MeterState>(optimumDirection === 'low' ? 'high' : 'optimum');
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: Addressed. For mid-optimum meters, the high segment now renders with the danger tone (`high`) instead of optimum, aligning the guide with the computed high-state semantics.

File: packages/components/src/components/meter/meter.svelte [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+    }
+    if (effectiveOptimum >= segmentHigh) {
+      if (clampedValue < segmentLow) return 'high';
+      if (clampedValue <= segmentHigh) return 'low';
+      return 'optimum';
+    }
+    if (clampedValue < segmentLow) return 'low';
+    if (clampedValue <= segmentHigh) return 'optimum';
+    return 'high';
+  });
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte:105]: Implemented. I adjusted the directional state comparisons at the segment edges so omitted-boundary cases can still reach best/worst states at range extremes, and added regression tests for both high-is-better and low-is-better max-edge scenarios.

File: packages/components/src/components/meter/meter.svelte [hunk 19]
```diff
@@ -0,0 +1,207 @@
...
+  const meterState = $derived.by<MeterState | undefined>(() => {
+    if (!hasThresholds) return undefined;
+    if (effectiveOptimum <= segmentLow) {
+      if (clampedValue <= segmentLow) return 'optimum';
+      if (clampedValue <= segmentHigh) return 'low';
+      return 'high';
+    }
+    if (effectiveOptimum >= segmentHigh) {
+      if (clampedValue < segmentLow) return 'high';
+      if (clampedValue <= segmentHigh) return 'low';
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/meter/meter.svelte]: I verified this path and added explicit regression coverage for `value == high` with `optimum == high` (and existing max-edge coverage with omitted `high`). The current logic treats the high boundary as reached in high-is-better mode, and the new test locks that behavior in.

File: .changeset/shiny-brooms-wave.md [hunk 20]
```diff
@@ -0,0 +1,5 @@
+---
+'@lostgradient/cinder': minor
+---
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply .changeset/shiny-brooms-wave.md:3]: Thanks for the catch. I rechecked against `origin/main`: `packages/components/package.json` is still `0.4.1` on both branches, so this PR does not introduce a manual version bump. The `.changeset` entry is intentionally left unconsumed so the release workflow can include it in the separate Version Packages PR.
